cmd/kubeadm/app/util/config/cluster_test.go

```diff
@@ -374,80 +354,84 @@ func TestGetNodeRegistration(t *testing.T) {
-			configMap: fakeConfigMap{
-				name: kubeadmconstants.KubeadmConfigConfigMap, // ClusterConfiguration from kubeadm-config.
-				data: map[string]string{},
-			},
-			expectedError: true,
+			name:        "no endpoint",
+			nodeName:    nodeName,
+			expectedErr: true,
 		},
 		{
-			name: "invalid v1beta1 - ClusterStatus without APIEndopoints",
-			configMap: fakeConfigMap{
-				name: kubeadmconstants.KubeadmConfigConfigMap, // ClusterConfiguration from kubeadm-config.
-				data: map[string]string{
-					kubeadmconstants.ClusterStatusConfigMapKey: string(cfgFiles["ClusterStatus_v1beta1_Without_APIEndpoints"]),
+			name:     "valid ipv4 endpoint",
+			nodeName: nodeName,
+			staticPod: &fakeStaticPod{
+				component: kubeadmconstants.KubeAPIServer,
+				annotations: map[string]string{
+					kubeadmconstants.KubeAPIServerAdvertiseAddressEndpointAnnotationKey: "1.2.3.4:1234",
 				},
 			},
-			expectedError: true,
+			expectedEndpoint: &kubeadmapi.APIEndpoint{AdvertiseAddress: "1.2.3.4", BindPort: 1234},
 		},
 		{
-			name: "valid v1beta2",
-			configMap: fakeConfigMap{
-				name: kubeadmconstants.KubeadmConfigConfigMap, // ClusterConfiguration from kubeadm-config.
-				data: map[string]string{
-					kubeadmconstants.ClusterStatusConfigMapKey: string(cfgFiles["ClusterStatus_v1beta2"]),
+			name:     "invalid ipv4 endpoint",
+			nodeName: nodeName,
+			staticPod: &fakeStaticPod{
+				component: kubeadmconstants.KubeAPIServer,
+				annotations: map[string]string{
+					kubeadmconstants.KubeAPIServerAdvertiseAddressEndpointAnnotationKey: "1.2.3::1234",
 				},
 			},
+			expectedErr: true,
 		},
 		{
-			name: "invalid v1beta2 - No ClusterStatus in kubeadm-config ConfigMap",
-			configMap: fakeConfigMap{
-				name: kubeadmconstants.KubeadmConfigConfigMap, // ClusterConfiguration from kubeadm-config.
-				data: map[string]string{},
+			name:     "valid ipv6 endpoint",
+			nodeName: nodeName,
+			staticPod: &fakeStaticPod{
+				component: kubeadmconstants.KubeAPIServer,
+				annotations: map[string]string{
+					kubeadmconstants.KubeAPIServerAdvertiseAddressEndpointAnnotationKey: "[::1]:1234",
+				},
 			},
-			expectedError: true,
+			expectedEndpoint: &kubeadmapi.APIEndpoint{AdvertiseAddress: "::1", BindPort: 1234},
 		},
 		{
-			name: "invalid v1beta2 - ClusterStatus without APIEndopoints",
-			configMap: fakeConfigMap{
-				name: kubeadmconstants.KubeadmConfigConfigMap, // ClusterConfiguration from kubeadm-config.
-				data: map[string]string{
-					kubeadmconstants.ClusterStatusConfigMapKey: string(cfgFiles["ClusterStatus_v1beta2_Without_APIEndpoints"]),
+			name:     "invalid ipv6 endpoint",
+			nodeName: nodeName,
+			staticPod: &fakeStaticPod{
+				component: kubeadmconstants.KubeAPIServer,
+				annotations: map[string]string{
+					kubeadmconstants.KubeAPIServerAdvertiseAddressEndpointAnnotationKey: "[::1:1234",
 				},
 			},
-			expectedError: true,
+			expectedErr: true,
```

Comment:
We need to verify that correct port numbers are used. Hence we should check that 0, negative port numbers and ones above 65535 are not successfully parsed.